### ISSUE TITLE
All Products Block: Buttons Size Inconsistent

### ISSUE THREAD

**Issue by zdenys** [woocommerce/woocommerce-blocks#3539]
## Describe the bug
For All Products Block, the size of the `Add to cart`/`Selected options` buttons is inconsistent and depends on the number of columns. Buttons become uncomfortably smaller when the number of columns is increased.

## To reproduce

1. Add All Products Block
2. Change the number of Columns to 3, 4, 5, 6 in the right-hand-side Layout Settings panel

## Expected behavior
The buttons size should be consistent and possibly not change at all as it happens on the traditional Shop page when changing the number of products per row in Customise > WooCommerce > Product Catalog

![](https://d.pr/i/uoDKN8+)
_Link: https://d.pr/i/uoDKN8_

## Screenshots

![](https://d.pr/i/8Q2OQV+)
_Link: https://d.pr/i/8Q2OQV_

### WordPress:
* Core version: 5.6
* WooCommerce version: 4.8
* WooCommerce Blocks version: 3.9

## Additional context
Reported by the user on the Forums https://wordpress.org/support/topic/buttons-and-text-becomes-smaller-on-product-blocks/ after update to version 3.9.

**Comment by nerrad** [comment]
Thanks for the report!

@mikejolley would this be related to work done in #3443?

**Comment by GeoJunkie** [comment]
Reported in 3566521-zd

**Comment by GeoJunkie** [comment]
I was able to narrow it down to this block:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/f8ee9daf762abfa8ff0c3b2155e06a03a48dd693/assets/js/base/components/product-list/style.scss#L78-L80

As a workaround, I targeted the specific button in CSS using the defaults that showed when unchecking the property in the browser inspector:

![](https://d.pr/i/1ltCLb+)
Image Link: https://d.pr/i/1ltCLb

```css
.theme-twentytwenty .wc-block-grid.has-6-columns .wc-block-grid__product, .wc-block-grid.has-6-columns .wc-block-grid__product {
    font-size: 1em;
}
```

**Comment by mikejolley** [comment]
@nerrad yes it is; as more columns are added, text wraps and buttons spill from the containers. This style was added so that more columns would decrease the font size. I think thats worse tbh — proper solution would be customisable font size for the products as a whole right?

**Comment by nerrad** [comment]
> I think thats worse tbh — proper solution would be customisable font size for the products as a whole right?

To clarify, are you saying this behaviour is worse:

> more columns would decrease the font size

If so, I think I agree. It's clear to me what you mean by "customisable font size for the products as a whole" though. Will you elaborate on that please?

**Comment by mikejolley** [comment]
Sorry, let me answer more clearly.

We added a style that would decrease the font size based on the number of columns used. That was done to mitigate (as best as possible) buttons being too large for the container and either spilling out of the container, or wrapping the text in an unreadable way. 

I think the smaller buttons appear better, and are easier to read, than the buttons we had previously which wrapped. This can be subjective based on the theme of course.

> If so, I think I agree. It's clear to me what you mean by "customisable font size for the products as a whole" though. Will you elaborate on that please?

I think we should have at block level a font size option. The font size would be applied to the parent element, and all children elements (the products in the grid) would inherit from this using em units.

**Comment by nerrad** [comment]
> I think the smaller buttons appear better, and are easier to read, than the buttons we had previously which wrapped. This can be subjective based on the theme of course.

Ahh, 👍 

> I think we should have at block level a font size option. The font size would be applied to the parent element, and all children elements (the products in the grid) would inherit from this using em units.

So his would be a block inspector control? If so, I think I agree and this sounds like something that would fit in with the Gutenberg Global styles work (in which case we may want to delay implementation).

So what do you suggest as the next steps for this issue Mike?

**Comment by mikejolley** [comment]
> So what do you suggest as the next steps for this issue Mike?

I don't think we should change or take action on this right now. I think we should revisit with the global styles work you mention, since that puts font size in the user's control.

Right now, custom CSS can increase the button size if the use case needs it.

**Comment by nerrad** [comment]
Thanks Mike, I agree.

**Comment by zdenys** [comment]
Another user report on the forums https://wordpress.org/support/topic/block-all-products-woocommerce/

**Comment by zdenys** [comment]
Another user report on the forums https://wordpress.org/support/topic/how-to-increase-the-font-size-price-and-button/
